### PR TITLE
raidboss: Convert Browser TTS Engine to TypeScript

### PR DESCRIPTION
**Pull request by panicstevenson**

### CODE REVIEW

File: ui/raidboss/browser_tts_engine.ts [hunk 1]
```diff
@@ -21,28 +25,44 @@ class SpeechTTSItem extends TTSItem {
...
 }
 
-class GoogleTTSItem extends TTSItem {
-  constructor(text, lang) {
-    super();
-    this.lang = lang;
+class GoogleTTSItem implements TTSItem {
+  readonly text: string;
+  readonly lang: string;
+  private item?: Element | null;
```
Comment (quisquous) [inline ui/raidboss/browser_tts_engine.ts:31]: (just for my own knowledge, does the `?` already imply can be null, or do you also have to say null?)

File: ui/raidboss/browser_tts_engine.ts [hunk 2]
```diff
@@ -96,13 +111,17 @@ export default class BrowserTTSEngine {
     }
   }
 
-  playSpeechTTS(text) {
-    this.ttsItems[text] = new SpeechTTSItem(text, this.speechLang, this.speechVoice);
-    this.ttsItems[text].play();
+  playSpeechTTS(text: string): void {
+    if (this.speechLang && this.speechVoice) {
+      const ttsItem = new SpeechTTSItem(text, this.speechLang, this.speechVoice);
```
Comment (quisquous) [inline ui/raidboss/browser_tts_engine.ts:116]: Up above, speechLang and speechVoice are optional, but here they're required.  What happens if these are undefined?

File: ui/raidboss/browser_tts_engine.ts (hunk 1 above)
Comment (quisquous) [reply ui/raidboss/browser_tts_engine.ts:31]: Or, looking at this again, when can it be null? Should it just be `private item?: Element`?

File: ui/raidboss/browser_tts_engine.ts (hunk 2 above)
Comment (panicstevenson) [reply ui/raidboss/browser_tts_engine.ts:116]: That's my fault for not updating the constructor; the constructor can be made to take lang and voice as optional parameters, and this check isn't necessary.

File: ui/raidboss/browser_tts_engine.ts (hunk 1 above)
Comment (panicstevenson) [reply ui/raidboss/browser_tts_engine.ts:31]: `?` implies `undefined`; in this case, `Document.createElement` returns `HTMLIFrameElement | null`. Removing the `| null` will return:
```
Type 'Element | null' is not assignable to type 'HTMLIFrameElement | undefined'.
  Type 'null' is not assignable to type 'HTMLIFrameElement | undefined'
  ```

**Pull request merged**